workers/unity/Packages/io.improbable.gdk.buildsystem/.codegen/Source/WorkerGenerationJob.cs

```diff
@@ -25,33 +24,23 @@ public WorkerGenerationJob(string baseOutputDir, IFileSystem fileSystem, Details
             const string jobName = nameof(WorkerGenerationJob);
             Logger.Info($"Initialising {jobName}.");
 
-            workerTypesToGenerate = ExtractWorkerTypes(CodeGeneratorOptions.Instance.WorkerJsonDirectory);
+            var workerTypesToGenerate = ExtractWorkerTypes(CodeGeneratorOptions.Instance.WorkerJsonDirectory);
 
-            var outputFilePaths = new List<string>
-            {
-                Path.Combine(relativeEditorPath, WorkerFileName),
-                Path.Combine(relativeEditorPath, WorkerListFileName),
-                Path.Combine(relativeOutputPath, BuildSystemFileName)
-            };
-
-            AddOutputFiles(outputFilePaths);
-            Logger.Info($"Added {outputFilePaths.Count} job output files.");
+            Logger.Trace($"Adding job target {WorkerFileName}.");
+            AddJobTarget(Path.Combine(relativeEditorPath, WorkerFileName),
+                () => UnityWorkerMenuGenerator.Generate(workerTypesToGenerate));
 
-            Logger.Info($"Finished initialising {jobName}.");
-        }
+            Logger.Trace($"Adding job target for {BuildSystemFileName}.");
+            AddJobTarget(Path.Combine(relativeEditorPath, BuildSystemFileName),
+                () => BuildSystemAssemblyGenerator.Generate());
 
-        protected override void RunImpl()
-        {
-            Logger.Info($"Generating {WorkerFileName}.");
-            var workerCode = UnityWorkerMenuGenerator.Generate(workerTypesToGenerate).Format();
-            AddContent(Path.Combine(relativeEditorPath, WorkerFileName), workerCode);
+            Logger.Trace($"Adding job target for {WorkerListFileName}.");
+            AddJobTarget(Path.Combine(relativeOutputPath, WorkerListFileName),
+                () => string.Join(Environment.NewLine, workerTypesToGenerate));
 
-            Logger.Info($"Generating {BuildSystemFileName}.");
-            var assemblyCode = BuildSystemAssemblyGenerator.Generate();
-            AddContent(Path.Combine(relativeOutputPath, BuildSystemFileName), assemblyCode);
+            Logger.Info("Added 3 job targets.");
 
-            Logger.Info($"Generating {WorkerListFileName}.");
-            AddContent(Path.Combine(relativeEditorPath, WorkerListFileName), string.Join(Environment.NewLine, workerTypesToGenerate));
+            Logger.Info($"Finished initialising {jobName}.");
```

Comment:
I think this and the earlier "initialising" log are too verbose. Should be either trace or just not printed?

Reply:
not printed - since we log this when using the activator in the EntryPoint anyway